pandas/io/common.py

```diff
@@ -583,12 +583,31 @@ def get_handle(
     Returns the dataclass IOHandles
     """
     # Windows does not default to utf-8. Set to utf-8 for a consistent behavior
-    encoding_passed, encoding = encoding, encoding or "utf-8"
+    encoding = encoding or "utf-8"
 
     # read_csv does not know whether the buffer is opened in binary/text mode
     if _is_binary_mode(path_or_buf, mode) and "b" not in mode:
         mode += "b"
 
+    # valdiate errors
+    if isinstance(errors, str):
+        errors = errors.lower()
+    if errors not in (
+        None,
+        "strict",
+        "ignore",
+        "replace",
+        "xmlcharrefreplace",
+        "backslashreplace",
+        "namereplace",
+        "surrogateescape",
+    ):
+        raise ValueError(
+            f"Invalide value for `encoding_errors` ({errors}). Please see "
```

Comment:
Is it okay to rename `get_handle`'s `errors` to `encoding_errors`? `get_handle` is technically not a private function.

Reply:
instead of enumerating all possible error codes, I could also use `codecs.lookup_error` and then raise a different error message. Would that be better?

Reply:
@jreback Do you have thoughts about using `codecs.lookup_error(errors)` to then either let it directly error (LookupError) or catch its error and raise a different error? Using `codecs.lookup_error` would make it more future-proof and it is also similar to how the new XML code detects bad `encodings` https://github.com/pandas-dev/pandas/blob/f290b65c82bae740cd039c21d5f660c797986d5f/pandas/io/formats/xml.py#L176

I could create a PR to call `codecs.lookup(encoding)` and `codecs.lookup_error(errors)` inside `get_handle` and if intended catch their errors and raise a more user-friendly error.